### PR TITLE
Adapt 2.0 evaluation to cross modal example

### PR DESCRIPTION
**Pull request by Kelton8Z**

### CODE REVIEW

File: cross-modal-search/pods/executors.py [hunk 1]
```diff
@@ -8,7 +8,17 @@
 from PIL import Image
 import io
 from jina import Executor, DocumentArray, requests, Document
-from jina.helloworld.chatbot.my_executors import _norm, _ext_B, _ext_A, _cosine
+from jina.helloworld.chatbot.executors import _norm, _ext_B, _ext_A, _cosine
+
+
+class ReciprocalRankEvaluator(Executor):
+    def rank_evaluate(self, actual: Sequence[Union[str, int]], desired: Sequence[Union[str, int]], **kwargs):
```
Comment (JoanFM) [inline cross-modal-search/pods/executors.py:15]: It needs the `request` decorator

Comment (JoanFM) [reply cross-modal-search/pods/executors.py:15]: and please review `Executor` cookbook, this will never work with 2.0

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: cross-modal-search/pods/executors.py (hunk 1 above)
Comment (Kelton8Z) [reply cross-modal-search/pods/executors.py:15]: there is request routing in yaml

Comment (Kelton8Z) [reply cross-modal-search/pods/executors.py:15]: the actual ranking and evaluating is still WIP

File: cross-modal-search/pods/evaluate.yml [hunk 2]
```diff
@@ -1,10 +1,8 @@
-!ReciprocalRankEvaluator
-with:
-  {}
+jtype: ReciprocalRankEvaluator
+#with:
+#  {}
```
Comment (bwanglzu) [inline cross-modal-search/pods/evaluate.yml:3]: unused comments

File: cross-modal-search/pods/evaluate.yml [hunk 3]
```diff
@@ -1,10 +1,8 @@
...
 requests:
-    on:
-      SearchRequest:
-        - !RankEvaluateDriver
-          with:
-            fields: ['tags__id']
-            traversal_paths: ['r']
\ No newline at end of file
+  /search: rank_evaluate
+#            traversal_paths: ['r']
```
Comment (bwanglzu) [inline cross-modal-search/pods/evaluate.yml:8]: unused comments

File: cross-modal-search/pods/executors.py [hunk 4]
```diff
@@ -11,10 +11,40 @@
...
+        :return: Reciprocal rank score
+        """
+        if len(actual) == 0 or len(desired) == 0:
+            return 0.0
+        try:
+            return 1.0 / (actual.index(desired[0]) + 1)
+        except:
+            return 0.0
+
+    @requests
```
Comment (bwanglzu) [inline cross-modal-search/pods/executors.py:35]: do we want to evaluate on `index` and `search` time?

File: cross-modal-search/evaluate.py [hunk 5]
```diff
@@ -58,6 +59,7 @@ def evaluation_generator(num_docs=None, batch_size=8, dataset_type='f8k', mode='
                     document.buffer = image
                     document.modality = 'image'
                     document.mime_type = 'image/jpeg'
+                    document.tags['id'] = hashed
```
Comment (bwanglzu) [inline cross-modal-search/evaluate.py:62]: why this is needed?

**Review by bwanglzu — CHANGES_REQUESTED** [review]
Some minor comments

File: cross-modal-search/pods/evaluate.yml (hunk 2 above)
Comment (Kelton8Z) [reply cross-modal-search/pods/evaluate.yml:3]: thx for pointing out, lemme clean it up

File: cross-modal-search/pods/executors.py (hunk 4 above)
Comment (Kelton8Z) [reply cross-modal-search/pods/executors.py:35]: no, just search time, thx for pointing out

File: cross-modal-search/evaluate.py (hunk 5 above)
Comment (Kelton8Z) [reply cross-modal-search/evaluate.py:62]: actually redundant

File: cross-modal-search/pods/executors.py (hunk 4 above)
Comment (Kelton8Z) [reply cross-modal-search/pods/executors.py]: although I believe `/search: rank_evaluate` in evaluate.yml also does the job

File: cross-modal-search/dataset.py [hunk 6]
```diff
@@ -116,9 +116,8 @@ def input_index_data(num_docs=None, batch_size=8, dataset_type='f30k'):
                 document_img.buffer = image
                 document_img.modality = 'image'
                 document_img.mime_type = 'image/jpeg'
-                document_img.tags['id'] = hashed
 
-            with Document() as document_caption:
+            with Document(id=hashed) as document_caption:
```
Comment (bwanglzu) [inline cross-modal-search/dataset.py:120]: what is the difference between `id` and `tags.id`?

**Comment by JoanFM** [comment]
@bwanglzu it is to showcase that id sometimes is random and out of users control and tags id u can always control

**Comment by bwanglzu** [comment]
okay, I remember now

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**